File: LeetCodePython/src/BestPokerHand.py
class Solution:
    def bestHand(self, ranks: List[int], suits: List[str]) -> str:
        suitRecord = {}
        rankRecord = {}
        for rank in ranks:
            count = rankRecord.setdefault(rank, 0)
            rankRecord.update({rank: count + 1})
        for suit in suits:
            count = suitRecord.setdefault(suit, 0)
            suitRecord.update({suit: count + 1})
        if suitRecord.__len__() == 1:
            return "Flush"
        max_count = 0
        for key, val in rankRecord.items():
            max_count = max(max_count, val)
        if max_count >= 3:
            return "Three of a Kind"
        elif max_count == 2:
            return "Pair"
        else:
            return "High Card"

File: LeetCodeJava/src/BestPokerHand.java
import java.util.HashMap;



class BestPokerHand {
    public static void main(String[] args) {
        int[] ranks = new int[5];
        ranks[0] = 4;
        ranks[1] = 4;
        ranks[2] = 2;
        ranks[3] = 4;
        ranks[4] = 4;
        char[] suits = new char[5];
        suits[0] = 'd';
        suits[1] = 'a';
        suits[2] = 'a';
        suits[3] = 'b';
        suits[4] = 'c';
        bestHand(ranks, suits);
        System.out.println( bestHand(ranks, suits));
    }

    public static String bestHand(int[] ranks, char[] suits) {
        HashMap<Integer, Integer> rankRecord = new HashMap<>();
        HashMap<Character, Integer> suitRecord = new HashMap<>();
        for (int rank : ranks) {
            rankRecord.put(rank, rankRecord.getOrDefault(rank, 1) + 1);
        }
        for (char suit : suits) {
            suitRecord.put(suit, suitRecord.getOrDefault(suit, 1) + 1);
        }
        if (suitRecord.size() == 5) {
            return "Flush";
        }
        int maxSameRank = 0;
        for (int rank : rankRecord.keySet()) {
            maxSameRank = Math.max(rankRecord.get(rank), maxSameRank);
        }
        if (maxSameRank >= 3) {
            return "Three of a Kind";
        } else if (maxSameRank == 2) {
            return "Pair";
        } else {
            return "High Card";
        }
    }
}